### PR TITLE
Add Elliptic bitcoin dataset

### PR DESCRIPTION
**Pull request by vinayhpandya**
## Added elliptic bitcoin transaction dataset 

Based on pytorch's version for elliptic bitcoin dataset
https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/datasets/elliptic.html

## Checklist

Put an `x` in the boxes that apply.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation

### CODE REVIEW

File: mlx_graphs/datasets/elliptic.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            "elliptic_txs_classes.csv",
+        ]
+
+    def download(self):
+        # This url is unable to download the data for elliptic bitcoin dataset
+        # lets try with pytorch geometric to verify the data
+        url = "https://data.pyg.org/datasets/elliptic/"
+        for files in self.raw_file_names:
+            download(f"{url}{files}.zip", self.raw_path)
+            extract_archive(f"{self.raw_path}/{files}.zip", self.processed_path)
```
Comment (francescofarina) [inline mlx_graphs/datasets/elliptic.py:67]: why are you extracting the data in `processed_path`? I guess it should go in `raw_path`

**Review by francescofarina — CHANGES_REQUESTED** [review]

**Comment by TristanBilot** [comment]
Amazing @vinayhpandya, thanks for adding this dataset! Apart from @francescofarina's comment, it seems all good to me.

File: mlx_graphs/datasets/elliptic.py (hunk 1 above)
Comment (vinayhpandya) [reply mlx_graphs/datasets/elliptic.py]: Updated in the new commit. I have also added pandas in the dependency section since the tests were failing

**Review by francescofarina — COMMENTED** [review]
Very up for discussion: looks like pandas is only used to parse the csv. Do we really need to depend on pandas? @vinayhpandya @TristanBilot wdyt?

**Comment by vinayhpandya** [comment]
@francescofarina I don't think its absolutely necessary to add pandas, I can try to see if numpy can extract csv values for nodes and edges and that should work as well

**Comment by francescofarina** [comment]
> @francescofarina I don't think its absolutely necessary to add pandas, I can try to see if numpy can extract csv values for nodes and edges and that should work as well

That would be great - I think it's best to not add `pandas` if `numpy` works as well!

**Comment by vinayhpandya** [comment]
I tried using numpy's loadtxt functionality to manipulate the node features but it takes around 9 secs
compared to 3 secs with Pandas, is csv reader a better option?
`Time taken is 9.910717964172363
Feature shape is (203769, 167)
Time taken for pandas is 3.2206032276153564`

**Comment by francescofarina** [comment]
@vinayhpandya I think the 9 seconds with numpy are ok. The dataset is processed only once and then stored in memory to be reloaded when needed, so that should be ok 👍🏻

**Comment by TristanBilot** [comment]
Good catch! It would be indeed better to use numpy here, even if slower.

**Comment by vinayhpandya** [comment]
@TristanBilot @francescofarina I have removed pandas and used numpy's `loadtxt` to process the files. Although this process is slightly slower than pandas (upto ~7secs)

**Pull request merged**